ratis-client/src/main/java/org/apache/ratis/client/impl/DataStreamClientImpl.java

```diff
@@ -110,7 +110,12 @@ public DataStreamClientRpc getClientRpc() {
 
   @Override
   public DataStreamOutput stream() {
-    return new DataStreamOutputImpl(streamId.incrementAndGet());
+    return stream(groupId);
+  }
+
+  @Override
+  public DataStreamOutput stream(RaftGroupId groupId) {
```

Comment:
Thanks! This is supposed to help RATIS-1085 multiple group support. Will rebase against this PR and continue my implementation for RATIS-1085